src/plugins/intel_cpu/src/nodes/mkldnn_gather_node.cpp

```diff
@@ -34,79 +39,184 @@ bool MKLDNNGatherNode::isSupportedOperation(const std::shared_ptr<const ov::Node
+
+    dataTypeSize = getOriginalInputPrecisionAtPort(GATHER_DATA).size();
 
-    if (op->get_input_node_shared_ptr(GATHER_AXIS)->get_type_info() == ov::op::v0::Constant::get_type_info_static()) {
+    if (ov::is_type<ov::op::v0::Constant>(op->get_input_node_ptr(GATHER_AXIS))) {
         isAxisInputConst = true;
         axis = ov::as_type<ov::op::v0::Constant>(op->get_input_node_ptr(GATHER_AXIS))->cast_vector<int>()[0];
         if (axis < 0)
             axis += dataSrcRank;
         if (axis < 0 || axis >= dataSrcRank || batchDims > axis)
-            IE_THROW() << errorPrefix << "has incorrect input parameter axis value: " << axis;
+            THROW_ERROR << "has incorrect input parameter axis value: " << axis;
+
+        if (isDataShapeStat) {
+            const auto& dataDims = dataShape.getDims();
+            axisDim = dataDims[axis];
+            beforeAxisSize = std::accumulate(dataDims.begin(), dataDims.begin() + axis, 1lu, std::multiplies<Dim>());
+            betweenBatchAndAxisSize = std::accumulate(dataDims.begin() + batchDims, dataDims.begin() + axis, 1lu, std::multiplies<Dim>());
+            afterAxisSize = std::accumulate(dataDims.begin() + axis + 1, dataDims.end(), 1lu, std::multiplies<Dim>());
+
+            afterAxisSizeInBytes = afterAxisSize * dataTypeSize;
+            axisAndAfterAxisSizeInBytes = axisDim * afterAxisSizeInBytes;
+            srcAfterBatchSizeInBytes = betweenBatchAndAxisSize * axisAndAfterAxisSizeInBytes;
+        }
+    }
+
+    if (isDataShapeStat) {
+        const auto& dataDims = dataShape.getDims();
+        beforeBatchSize = std::accumulate(dataDims.begin(), dataDims.begin() + batchDims, 1lu, std::multiplies<Dim>());
+    }
+    if (isIdxShapeStat) {
+        const auto& idxDims = idxShape.getDims();
+        specIndicesSize = std::accumulate(idxDims.begin() + batchDims, idxDims.end(), 1lu, std::multiplies<Dim>());
+
+        if (isDataShapeStat) {
+            specIdxAndAfterAxSizeB = specIndicesSize * afterAxisSizeInBytes;
+            totalWork = beforeBatchSize * betweenBatchAndAxisSize * specIndicesSize * afterAxisSize;
+        }
     }
-    dataSize = getOriginalInputPrecisionAtPort(GATHER_DATA).size();
 }
 
 void MKLDNNGatherNode::initSupportedPrimitiveDescriptors() {
     if (!supportedPrimitiveDescriptors.empty())
         return;
 
+    auto typeCandidate = ref_any, type = ref_any;
+    size_t vlen = 1;
+    if (x64::mayiuse(x64::avx512_common)) {
+        typeCandidate = jit_avx512;
+        vlen = x64::cpu_isa_traits<x64::avx512_common>::vlen / dataTypeSize;
+    } else if (x64::mayiuse(x64::avx2)) {
+        typeCandidate = jit_avx2;
+        vlen = x64::cpu_isa_traits<x64::avx2>::vlen / dataTypeSize;
+    }
+
+    if (typeCandidate != ref_any) {
+         if (isDynamicNode()) {
+            if (isDataShapeStat && isAxisInputConst && afterAxisSize == 1) {
+                type = typeCandidate;
+            }
+         } else if (afterAxisSize < vlen) {
```

Comment:
1. Is it major technical limitation that JIT impl cannot support afterAxisSize >= vlen cases? 
2. Do we have understanding how many ops from benchmarking suite are not covered by JIT impl?

Reply:
1. Significant bottleneck is case with afterAxisSize == 1. This case is totally covered by the kernel.
1 < afterAxisSize < vlen is extension for some cases.
afterAxisSize > vlen - is not covered due to memcpy from reference shows good performance here and I suppose kernel will not produce significant gain.
2. Covered cases gives about 5% for a few models. Most others shows about 2-3%. I'll try to provide detailed report later.